### ISSUE TITLE
Calling stream.destroy(error) kills node

### ISSUE THREAD

**Issue by alxndrsn** [gajus/slonik#378]
## Expected Behavior

Should be able to call `stream.destroy(error)` without killing node process.

See: https://nodejs.org/api/stream.html#readabledestroyerror

## Current Behavior

Calling `stream.destroy(error)` kills node process.

## Possible Solution

https://github.com/gajus/slonik/pull/360

## Steps to Reproduce

Failing test at https://github.com/gajus/slonik/pull/377

## Logs

```
  Uncaught exception in test/slonik/integration/pg.ts

  test/slonik/integration/pg.ts:347

   346:   `, (stream) => {                   
   347:     stream.destroy(new Error('Foo'));
   348:   });                                

  Error: Foo

  › test/slonik/integration/pg.ts:347:20
  › src/connectionMethods/stream.ts:4:205
  › src/connectionMethods/stream.ts:2:1732
  › executeQuery (src/routines/executeQuery.ts:13:23)
  › async stream (src/connectionMethods/stream.ts:2:1009)
  › async createConnection (src/factories/createConnection.ts:10:21)
  › async test/slonik/integration/pg.ts:344:3

  ✖ test/slonik/integration/pg.ts exited with a non-zero exit code: 1
```

## Related

* https://github.com/gajus/slonik/issues/347

**Comment by alxndrsn** [comment]
Perhaps this is deliberate behaviour.

I've extended the test from #377 at #379, and instead get the output:

```
  ✖ Timed out while running tests

  1 tests were pending in test/slonik/integration/pg.ts

  ◌ integration › pg › does not crash after destroying a stream with an error, and can make more requests
```

**Comment by alxndrsn** [comment]
Fixed in v31.2.3.
